flink-runtime/src/main/java/org/apache/flink/runtime/taskmanager/Task.java

```diff
@@ -530,8 +539,10 @@ public void startTaskThread() {
 	@Override
 	public void run() {
 		try {
+			MDC.put("jobName", this.jobName);
```

Comment:
We may want to introduce an opt-in configuration option. I cannot assess how many threads are being created in existing jobs, and the copy of the context map makes be a bit wary.

Reply:
Maybe this should also be configurable to use the job ID.